cardano-testnet/src/Testnet/Cardano.hs

```diff
@@ -298,98 +378,134 @@ cardanoTestnet testnetOptions H.Conf {..} = do
 
   -- Create keys and addresses to withdraw the initial UTxO into
-  forM_ bftNodesN $ \n -> do
-    execCli_
-      [ "keygen"
-      , "--secret", tempAbsPath </> "byron/payment-keys.00" <> show @Int (n - 1) <> ".key"
-      ]
-
-    H.execCli
-      [ "signing-key-address"
-      , "--testnet-magic", show @Int testnetMagic
-      , "--secret", tempAbsPath </> "byron/payment-keys.00" <> show @Int (n - 1) <> ".key"
-      ] >>= H.writeFile (tempAbsPath </> "byron/address-00" <> show @Int (n - 1))
-
-    -- Write Genesis addresses to files
-    H.execCli
-      [ "signing-key-address"
-      , "--testnet-magic", show @Int testnetMagic
-      , "--secret", tempAbsPath </> "byron/genesis-keys.00" <> show @Int (n - 1) <> ".key"
-      ] >>= H.writeFile (tempAbsPath </> "byron/genesis-address-00" <> show @Int (n - 1))
-
-  do
-    richAddrFrom <- S.firstLine <$> H.readFile (tempAbsPath </> "byron/genesis-address-000")
-    txAddr <- S.firstLine <$> H.readFile (tempAbsPath </> "byron/address-000")
+  byronAddress0:_ <- forM [1 .. numBftNodes] $ \n -> do
+    skey <- cliKeyGen tmpDir $ "byron/payment-keys.00" <> show @Int (n - 1) <> ".key"
+    cliSigningKeyAddress tmpDir testnetMagic skey $ "byron/address-00" <> show @Int (n - 1)
+
+  byronGensisAddress0:_ <- forM (zip [1 .. numBftNodes] genesisKeys) $ \(n,key) -> do
+    cliSigningKeyAddress tmpDir testnetMagic
+      key
+      ("byron/genesis-address-00" <> show @Int (n - 1))
+
+  richAddrFrom <- S.firstLine <$> H.readFile (tmpDir </> "byron/genesis-address-000")
+  txAddr <- S.firstLine <$> H.readFile (unFile byronAddress0)
+  let
+      maxByronSupply = 10020000000
+      fundsPerGenesisAddress = maxByronSupply `div` numBftNodes
+      fundsPerByronAddress = fundsPerGenesisAddress - 100000000
 
-    -- Create Byron address that moves funds out of the genesis UTxO into a regular
-    -- address.
-    execCli_
-      [ "issue-genesis-utxo-expenditure"
-      , "--genesis-json", tempAbsPath </> "byron/genesis.json"
-      , "--testnet-magic", show @Int testnetMagic
-      , "--tx", tempAbsPath </> "tx0.tx"
-      , "--wallet-key", tempAbsPath </> "byron/delegate-keys.000.key"
-      , "--rich-addr-from", richAddrFrom
-      , "--txout", show @(String, Word64) (txAddr, fundsPerByronAddress)
-      ]
+  -- Create Byron address that moves funds out of the genesis UTxO into a regular
+  -- address.
+  execCli_
+    [ "issue-genesis-utxo-expenditure"
+    , "--genesis-json", tempAbsPath </> "byron/genesis.json"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--tx", tempAbsPath </> "tx0.tx"
+    , "--wallet-key", tempAbsPath </> "byron/delegate-keys.000.key"
+    , "--rich-addr-from", richAddrFrom
+    , "--txout", show @(String, Int) (txAddr, fundsPerByronAddress)
+    ]
 
-  -- Update Proposal and votes
+{-
+cardanoTestnetByronGovernance creates two update-proposals and two update-votes.
+However they are never transmitted to the chain ?!
```

Comment:
These were translated directly from the `mkfiles.sh` script this was based on.  Those scripts were designed so that the update proposals can be submitted manually.

We should consider what we want to do with this in the Haskell code.

Reply:
Good information.
I should have a look at `mkfiles.sh`. Is it still available ? Is it still in use ?
I'll change that comment to a TODO:
There should be a test that submits the update proposals and checks that they are activated.
It was not clear to me, if that test had been removed or disabled, or if it had never existed.